plugins/inertia.md

```diff
@@ -0,0 +1,249 @@
+
+```html
+<script setup lang="ts">
+  import { useForm } from '@formkit/inertia'
+
+  const form = useForm()
+  form.on('before', () => {
+    return confirm('Are you sure you want to delete this user?')
+  })
+</script>
+```
+
+The `combine()` function is just a easier way to add multiple events in a single place:
+
+```html
+<script setup lang="ts">
+  import { useForm } from '@formkit/inertia'
+
+  const form = useForm()
+  form.combine((on) => {
+    on('before', () => {
+      return confirm('Are you sure you want to delete this user?')
+    })
+
+    on('success', () => {
+      toast('User deleted.')
+    })
+  })
+</script>
+```
+
+## Event System
+
+The core functionality of all comes from the simple and yet porweful event system wrapper for Inertia visit options, the `useForm()` uses the `useEventsSystem()` to create a way to access and retrive the event callbacks with multiple function calls correctly, without having to deal with object merging.
+
+The `useEventsSystem()` returns 4 functions `on()`, `combine()`, `execute()` and `toVisitOptions()`, the `on` function is just the same as its passed to the return of `useForm`, it also accepts these events `before`, `start`, `progress`, `success`, `error`, `cancel`, `finish`:
+
+```ts
+const event = useEventsSystem()
+event.on('before', (visit) => {
+  console.log(visit)
+})
+```
+
+As you can see it only gets `visit` as a parameter because `useEventsSystem()` was not specified that its events will receive more than that, but you can extend by passing an array of types of parameters to it:
+
+```ts
+const event = useEventsSystem<[node: FormKitNode]>()
+event.on('before', (visit, node) => {
+  console.log(visit, node)
+})
+```
+
+The `combine()` function is meant to be used for easily pass multiple events in a single place:
+
+```ts
+// addon.ts
+return (on) => {
+  on('before', (visit, node) => {
+    console.log(visit, node)
+  })
+
+  on('success', (page, node) => {
+    console.log(page, node)
+  })
+}
+
+// app.ts
+import addon from './addon'
+
+const event = useEventsSystem<[node: FormKitNode]>()
+event.combine(addon)
+```
+
+The `execute()` function runs the events expects the event and the parameters to returns the expected return of that event callback from Inertia:
+
+```ts
+const event = useEventsSystem<[node: FormKitNode]>()
+
+event.on('before', (visit, node) => {
+  console.log(visit, node)
+})
+event.on('before', (visit, node) => {
+  return false
+})
+
+const result = event.execute('before', visit, node) // runs console.log
+console.log(result) // returns false
+```
+
+The `toVisitOptions()` functions returns a `VisitOptions` with all events that where passed prior to it by wrapping the `execute()` function, and its already ready to used in Inertia's `router()` function:
```

Comment:
I don’t have a specific suggestion for this line because I dont think I understand it, but it sounds like a run-on. Maybe you can refactor this into a couple smaller sentences to help clear it up?